gph/src/ripser.h

```diff
@@ -1432,48 +1435,71 @@ class ripser
-                                birth_edge.first, birth_edge.second,
-                                death_edge.first, death_edge.second};
+                        // Infinite death indicates an essential bar in disguise
+                        value_t death = get_diameter(pivot);
+                        is_essential =
+                            (death == std::numeric_limits<value_t>::infinity());
+                        if (!is_essential) {
+                            /* Pairs should only be extracted if insertion was
+                             * first one! */
+                            size_t location = last_diameter_index++;
+                            diameters[location] = death;
+                            auto first_ins =
+                                deaths
+                                    .insert({get_index(get_entry(pivot)), location})
+                                    .second;
+
+                            /* Only insert when it is the first time this bar is
+                             * encountered
+                             */
+                            if (first_ins) {
+                                if (return_flag_persistence_generators) {
+                                    // Inessential flag persistence generators in dim >
+                                    // 0
+                                    std::vector<index_t> vertices_birth(dim + 1);
+                                    std::vector<index_t> vertices_death(dim + 2);
+
+                                    index_t birth_idx =
+                                        get_index(get_entry(column_to_reduce));
+                                    index_t death_idx = get_index(get_entry(pivot));
+
+                                    get_simplex_vertices(birth_idx, dim, n,
+                                                         vertices_birth.rbegin());
+                                    get_simplex_vertices(death_idx, dim + 1, n,
+                                                         vertices_death.rbegin());
+
+                                    edge_t birth_edge =
+                                        get_youngest_edge_simplex(vertices_birth);
+                                    edge_t death_edge =
+                                        get_youngest_edge_simplex(vertices_death);
+
+                                    finite_generator[location] = {
+                                        birth_edge.first, birth_edge.second,
+                                        death_edge.first, death_edge.second};
+                                }
+                            }
                         }
 
                         break;
                     }
                 } else {
+                    is_essential = true;
+
+                    // TODO: these will need special attention, if output
+                    // happens after the reduction, not during
+                    break;
```

Comment:
I never really understood this comment but I wonder if it still makes sense in this new position